build.sbt

```diff
@@ -271,6 +280,23 @@ lazy val connectCommon = (project in file("spark-connect/common"))
       PB.gens.java -> (Compile / sourceManaged).value,
       PB.gens.plugin("grpc-java") -> (Compile / sourceManaged).value
     ),
+    Compile / packageBin := assembly.value,
+    (assembly / test) := { },
+    (assembly / logLevel) := Level.Info,
+    // Exclude `scala-library` from assembly.
+    (assembly / assemblyPackageScala / assembleArtifact) := false,
+    (assembly / assemblyShadeRules) := Seq(
+      ShadeRule.rename("com.google.**" -> "org.deltaproject.connect.com.google.@1").inAll,
+      ShadeRule.rename("io.grpc.**" -> "org.deltaproject.connect.io.grpc.@1").inAll,
+      ShadeRule.rename("org.checkerframework.**" -> "org.deltaproject.connect.checkerframework.@1").inAll,
+      ShadeRule.rename("javax.annotation.**" -> "org.deltaproject.connect.javax.annotation.@1").inAll,
+    ),
```

Comment:
Could you extract these four rules as a variable to avoid code duplication?

Reply:
Nice catch, `sbt` file is still Scala code at the end of the day I guess.